api/settings/v1/metrics.proto

```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
+
+package settings.v1;
+
+import "types/v1/types.proto";
+
+message RecordingRuleSetting {
+  repeated RecordingRule rules = 1;
+}
+
+message RecordingRule {
+  // Unique id of the recording rule within the tenant.
+  string id = 1;
+  repeated string matchers = 2;
+  repeated string group_by = 3;
+  repeated types.v1.LabelPair external_labels = 4;
```

Comment:
> * profile type will be set in `matchers`
> 
> * service name will be set in `matchers`
> 
> * metric name can be set in `external_labels`
> 
> * keep_labels / labels_to_export is renamed to `group_by`

It makes sense to put them as comments into the `proto` file, maybe with an example value each.